qiskit/transpiler/passes/layout/disjoint_utils.py

```diff
@@ -127,6 +127,21 @@ def combine_barriers(dag: DAGCircuit, retain_uuid: bool = True):
                 node.op.label = None
 
 
+def check_layout_isolated_to_component(dag: DAGCircuit, coupling_map: CouplingMap) -> bool:
+    """Check that the layout of the dag does not require connectivity across connected components
+    in the CouplingMap"""
+    qubit_indices = {bit: index for index, bit in enumerate(dag.qubits)}
+    component_sets = [set(x.graph.nodes()) for x in coupling_map.connected_components()]
+    for inst in dag.two_qubit_ops():
```

Comment:
Do we need special handling at all for >2Q barriers?

Reply:
This should be handled already by the method call. This `DAGCircuit` method returns a list of op nodes with exactly 2 qubit operations and it filters out operations flagged as a `directive` which include barrier. So there shouldn't be anything needed here to account for a 2q barrier.